pkg/runtime/pubsub/outbox.go

```diff
@@ -177,12 +179,16 @@ func (o *outboxImpl) PublishInternal(ctx context.Context, stateStore string, ope
 
 				if proj.ContentType != nil {
 					contentType = *proj.ContentType
+				} else if ct, ok := proj.Metadata[metadata.ContentType]; ok {
```

Comment:
I believe this specific case was missed, but the original PR was merged into the `release-1.14` branch. Mind switching this PR branch target to point to the `release-1.14` branch and I think the bottom `if` diff can be removed since it is already in the `release-1.14` branch. 

By targeting our `release-1.14` branch this change will be in the next version `1.14.5` for you to test sooner. From there the commits will get cherry-picked back to master :)